lib/parse-message.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+module.exports = exports = {};
+
+
+exports.nickCommand = function(message, socket) {
+  console.log('username: ' + socket.username);
+  socket.username = message.split(' ').slice(1).join(' ').trim();
+  socket.write(`you are now ${socket.username}\n`);
+
+};
+
+exports.dmCommand = function(message, array){
+ // "/dm user_30 hello how are you"
+  let toUsername = message.split(' ')[1];
+  let content = message.split(' ').slice(2).join(' ').trim();
+  array.forEach(s => {
+    if (s.username === toUsername)
+      s.write(content);
+  });
+};
+
+exports.users = function(array, socket){
+  array.forEach(user => {
+    socket.write(user.username + '\n');
+  });
+};
+
+
+exports.trolololol = function (message, array) {
+  let content = message.split(' ').slice(1).join(' ').trim();
+  array.forEach(user => {
+    for (var i = 0; i < 10; i++) {
+      user.write(content + '\n');
+    }
+  });
+};
+
+exports.ban = function(message, array){
```

Comment:
Remember that the ``ban`` command also needs to remove the user from the ``sockets`` array in your server file. Look in your server file to find a similar method that Duncan already wrote to delete a user when they log out. You can just repeat that same logic here.